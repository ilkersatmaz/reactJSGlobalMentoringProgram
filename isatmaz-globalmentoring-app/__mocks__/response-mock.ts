export const movieResponse = {
	data: {
		id: 221122,
		title: 'first movie of the year',
		overview:
			'overview of the video for the m ',		
		genres: ['Action', 'Horror'],
		budget: 1000,
		poster_path: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
		release_date: '2023-01-01',
		revenue: 136907000,
		runtime: 134,
		tagline: "new and best",
		
		vote_average: 9.9,
		vote_count: 1145,
	},

	totalAmount: 1,
};
