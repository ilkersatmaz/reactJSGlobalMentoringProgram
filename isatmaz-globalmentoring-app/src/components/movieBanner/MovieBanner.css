.movie-banner-div{
    display: flex;
    flex-direction: row;
    background-color: #232323;
    padding: 90px 60px 20px 60px;
    margin-bottom: 20px;
}
.movie-detail-image{
    width: 30%;
    height: -webkit-fill-available;
    max-height: 486px;
}
.movie-detail-div{
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    margin-left: 60px;
}
.movie-detail-title{
    color: #ffffff;
    font-size: 40px;
    margin: 0;
}
.movie-detail-releaseDate{
    font-size: 24px;
    color: #F65261;
    height: fit-content;
    margin-right: 40px;
}
.movie-detail-genre-div{
    display: flex;
}
.movie-detail-genre{
    color: #ffffff;
    font-size: 14px;
    margin-left: 5px;
}
.movie-detail-description{
    color: #ffffff;
    font-size: 20px;
}

.movie-detail-rate{ 
    margin-left: 20px;
    color: #ffffff;
    border: 1px solid #FFFFFF;
    font-size: 20px;
    border-radius: 56%;
    padding: 10px;
    margin-top: 0;
    margin-bottom: 0;
}