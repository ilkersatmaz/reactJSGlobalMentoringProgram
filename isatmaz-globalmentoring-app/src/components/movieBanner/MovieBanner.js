import React from "react";
import 'reactjs-popup/dist/index.css';

import "./MovieBanner.css";

function MovieBanner(props){
    const {movieData}=props;

    function getYear(releaseDate){
        if(releaseDate){
            const date=new Date(releaseDate)
            return date.getFullYear();
        }
        return "";
    }

    function getDuration(runTime){
        if(runTime){
            const parts=runTime.split(":");
            if(parts && parts.length > 2){
                return parts[0] + "h " + parts[1] + "min" 
            }
        }
        return "";
    }

    return (
        <div className="movie-banner-div">
            <img alt={movieData.title} className="movie-detail-image" src={movieData.imageUrl} />
            <div className="movie-detail-div">
                <div className="movie-detail-genre-div">
                    <p className="movie-detail-title">{movieData.title}</p>    
                    <p className="movie-detail-rate">{movieData.rating}</p>  
                 </div>  
                
                <div className="movie-detail-genre-div">
                    {movieData.genre.map((genre, index)=> {
                        return <p key={genre} className="movie-detail-genre">{genre}{ (index===movieData.genre.length-1 ? '' : ',')}</p>
                    })}
                 </div>   
                 <div className="movie-detail-genre-div">
                    <p className="movie-detail-releaseDate">{getYear(movieData.releaseDate)}</p>  
                    <p className="movie-detail-releaseDate">{getDuration(movieData.runTime)}</p>  
                 </div>            
                 <p className="movie-detail-description">{movieData.description}</p>            
            </div>       
        </div>      
    )
}

export default MovieBanner