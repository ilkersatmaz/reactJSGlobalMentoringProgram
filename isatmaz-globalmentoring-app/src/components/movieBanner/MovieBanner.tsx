import React from "react";
import 'reactjs-popup/dist/index.css';

import "./MovieBanner.css";

function MovieBanner(props : any){
    const {movieData}=props;

    function getYear(releaseDate : string){
        if(releaseDate){
            const date=new Date(releaseDate)
            return date.getFullYear();
        }
        return "";
    }

    function getDuration(runTime : string){
        if(runTime){
            const {hours , minutes} =toHoursAndMinutes(runTime);           
            return hours + "h " + minutes + "min";            
        }
        return "";
    }

    function toHoursAndMinutes(totalMinutes) {
        const hours = Math.floor(totalMinutes / 60);
        const minutes = totalMinutes % 60;
      
        return { hours, minutes };
      }

    return (
        <div className="movie-banner-div">
            <img alt={movieData.title} className="movie-detail-image" src={movieData.poster_path} />
            <div className="movie-detail-div">
                <div className="movie-detail-genre-div">
                    <p className="movie-detail-title">{movieData.title}</p>    
                    <p className="movie-detail-rate">{movieData.vote_average}</p>  
                 </div>  
                
                <div className="movie-detail-genre-div">
                    {movieData.genres.map((genre : string, index : number)=> {
                        return <p key={genre} className="movie-detail-genre">{genre}{ (index===movieData.genres.length-1 ? '' : ',')}</p>
                    })}
                 </div>   
                 <div className="movie-detail-genre-div">
                    <p className="movie-detail-releaseDate">{getYear(movieData.release_date)}</p>  
                    <p className="movie-detail-releaseDate">{getDuration(movieData.runtime)}</p>  
                 </div>            
                 <p className="movie-detail-description">{movieData.overview}</p>            
            </div>       
        </div>      
    )
}

export default MovieBanner