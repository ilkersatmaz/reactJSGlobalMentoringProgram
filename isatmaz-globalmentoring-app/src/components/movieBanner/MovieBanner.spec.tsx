import { render } from '@testing-library/react';
import React from 'react';
import { movieResponse } from '../../../__mocks__/response-mock';
import MovieBanner from './MovieBanner';
describe('MovieBanner component', () => {
  it("should match the snapshot", () => { 
    const { asFragment } = render(<MovieBanner movieData={movieResponse.data}/>);
    
    expect(asFragment).toMatchSnapshot();
  });
})