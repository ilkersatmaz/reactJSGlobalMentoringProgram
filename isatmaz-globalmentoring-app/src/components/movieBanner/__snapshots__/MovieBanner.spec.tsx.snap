// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieBanner component should match the snapshot 1`] = `[Function]`;
